core/encoding.go

```diff
@@ -245,51 +244,33 @@ func DecodeIndexDataStoreKey(
 
 		i := len(key.Fields)
 		descending := false
+		var kind client.FieldKind = client.FieldKind_DocID
 		// If the key has more values encoded then fields on the index description, the last
 		// value must be the docID and we treat it as a string.
+		if i < len(fields) {
```

Comment:
suggestion: This check seems unnecessary - did something break without it?  How could it ever happen?

Something would have to be very very broken for this to ever be false no?

I suggest removal of the `if`.

todo: The comment above has been moved away from the line upon which it is commenting, please move it back :)

Reply:
no, nothing breaks. The two slices should have the same len.
I just thought some abnormal scenario (that I can't think of) these might be different.
But I think you are right, it is not necessary. I bought it back.